apps/admin_api/lib/admin_api/v1/controllers/tranasction_calculation_controller.ex

```diff
@@ -0,0 +1,57 @@
+defmodule AdminAPI.V1.TransactionCalculationController do
+  @moduledoc """
+  The controller to serve transaction calculations.
+  """
+  use AdminAPI, :controller
+  import AdminAPI.V1.ErrorHandler
+  alias EWallet.Exchange
+  alias EWalletDB.Token
+
+  @doc """
+  Calculates transaction amounts.
+  """
+  def calculate(conn, attrs) do
+    from_amount = attrs["from_amount"]
+    from_token = Token.get(attrs["from_token_id"])
+    to_amount = attrs["to_amount"]
+    to_token = Token.get(attrs["to_token_id"])
+
+    case do_calculate(from_amount, from_token, to_amount, to_token) do
```

Comment:
You can pass `attrs["from_amount"]` and `attrs["to_amount"] directly I think.

Reply:
Ah yes, i tried to cram all 4 params in but that looked very cluttered. Your suggestion looks like a good compromise. Updated.